doc/sphinx/proof-engine/ltac.rst

```diff
@@ -52,127 +66,98 @@ the variable :token:`ident` matches any complex expression with (possible)
 dependencies in the variables :n:`@ident__i` and returns a functional term
 of the form :n:`fun @ident__1 … ident__n => @term`.
 
-The main entry of the grammar is :n:`@ltac_expr`. This language is used in proof
-mode but it can also be used in toplevel definitions as shown below.
+.. note::
+
+   - If you want :n:`@tactic__2; @tactic__3` to be fully applied to the first
+     subgoal generated by :n:`@tactic__1` before applying it to the other
+     subgoals, then you should write:
+
+     - :n:`@tactic__1; [> @tactic__2; @tactic__3 .. ]` rather than
+
+     - :n:`@tactic__1; (@tactic__2; @tactic__3)`.
+
+   .. todo PR I tried to improve the wording for the following bullet,
+      but I'm not quite sure what it refers to.  Perhaps a fool's errand.
+      Does this issue still come up, and if so, exactly which productions?
+
+   - In :token:`tactic_arg_compat`, there is a grammatical ambiguity in
+     how to interpret :token:`qualid`.  It could be either a tactic argument
+     or just a :token:`term`.  This is resolved by checking first whether there
+     is a tactic by that name, then whether it is a reference to a term and
+     using the first one that's found.
+
+     The default interpretation can be overridden by using :n:`ltac:(@qualid)` or
+     :n:`(@qualid)` to force interpretation as, respectively, a tactic or a term.
+
+     .. todo PR With the described behavior, it would never be necessary to use
+        the "ltac:(@qualid)" form since it's checked first.
+
+.. insertprodn ltac_expr tactic_atom
+
+.. prodn::
+   ltac_expr ::= {| @ltac_expr4 | @binder_tactic }
+   let_clause ::= @name := @ltac_expr
+   | @ident {+ @name } := @ltac_expr
+   ltac_expr4 ::= @ltac_expr3 ; {| @ltac_expr3 | @binder_tactic }
+   | @ltac_expr3 ; [ {? @multi_goal_tactics } ]
+   | @ltac_expr3 ; [> {? @multi_goal_tactics } ]
```

Comment:
I don't think that's true; I think if you include the `;`, the tactic expression before it is mandatory.  However, the tactic expression+; as a chunk is optional

Reply:
I'm not sure I follow.

What I was saying is just that the first rule of `ltac_expr4` allows for `ltac_expr3; ltac_expr3` (assuming the precedence levels are correct) and that latter `ltac_expr3` can itself be a `[> ... ]`. What did I miss?

Reply:
I think I was replying to an earlier version of it before the edit, where, iirc, it was just claimed that the expr to the left of the semicolon was redundant

Reply:
> This rule is redundant

Yes, redundant because ltac_expr3 -> ...  -> ltac_expr0 -> multi_goal_tactics.

What is the difference between `[ {? @multi_goal_tactics } ]` and `[> {? @multi_goal_tactics } ]`?  There was no explanation in the text--would be nice to add.  Also I wonder why that's not available as part of ltac_expr0.

Reply:
Only `[>` is supported at top-level. `[` is supported only after `;`. I thought this was the only difference but apparently it isn't. Cf. the issue I opened to understand the difference better: #12283

Reply:
I've explained the difference as I understand it at https://github.com/coq/coq/issues/12283#issuecomment-626208208 (from the point of view of a frequent user of both `; [ .. ]` and `[ > .. ]`)

Reply:
I've also added a historical perspective at https://github.com/coq/coq/issues/12283#issuecomment-626210310

Reply:
I added a todo linking back to here so this can be addressed in more detail later. It would take more time than I should spend in this PR to unpack the details and draft appropriate wording.  The text includes the one-sentence definition.